Formula/pipenv.rb

```diff
@@ -0,0 +1,121 @@
+  end
+
+  resource "pathlib" do
+    url "https://files.pythonhosted.org/packages/ac/aa/9b065a76b9af472437a0059f77e8f962fe350438b927cb80184c32f075eb/pathlib-1.0.1.tar.gz"
+    sha256 "6940718dfc3eff4258203ad5021090933e5c04707d5ca8cc9e73c94a7894ea9f"
+  end
+
+  resource "pew" do
+    url "https://files.pythonhosted.org/packages/a4/be/efe77f7dcb088f263bfc5eac09121b2f9a6e59c3bdfdff7a801d2253f2ac/pew-1.1.0.tar.gz"
+    sha256 "d0f9d127cf321c959bb857e59080f7eaa59c8bbeb2d53aa07d3350aae50e1f2d"
+  end
+
+  resource "pycodestyle" do
+    url "https://files.pythonhosted.org/packages/e1/88/0e2cbf412bd849ea6f1af1f97882add46a374f4ba1d2aea39353609150ad/pycodestyle-2.3.1.tar.gz"
+    sha256 "682256a5b318149ca0d2a9185d365d8864a768a28db66a84a2ea946bcc426766"
+  end
+
+  resource "pyflakes" do
+    url "https://files.pythonhosted.org/packages/5b/b7/dcd6ebc826065ca4ccd2406aac4378e1df6eb91124625d45d520219932a1/pyflakes-1.5.0.tar.gz"
+    sha256 "aa0d4dff45c0cc2214ba158d29280f8fa1129f3e87858ef825930845146337f4"
+  end
+
+  resource "requests" do
+    url "https://files.pythonhosted.org/packages/b0/e1/eab4fc3752e3d240468a8c0b284607899d2fbfb236a56b7377a329aa8d09/requests-2.18.4.tar.gz"
+    sha256 "9c443e7324ba5b85070c4a818ade28bfabedf16ea10206da1132edaa6dda237e"
+  end
+
+  resource "shutilwhich" do
+    url "https://files.pythonhosted.org/packages/66/be/783f181594bb8bcfde174d6cd1e41956b986d0d8d337d535eb2555b92f8d/shutilwhich-1.1.0.tar.gz"
+    sha256 "db1f39c6461e42f630fa617bb8c79090f7711c9ca493e615e43d0610ecb64dc6"
+  end
+
+  resource "urllib3" do
+    url "https://files.pythonhosted.org/packages/ee/11/7c59620aceedcc1ef65e156cc5ce5a24ef87be4107c2b74458464e437a5d/urllib3-1.22.tar.gz"
+    sha256 "cc44da8e1145637334317feebd728bd869a35285b93cbb4cca2577da7e62db4f"
+  end
+
+  resource "virtualenv" do
+    url "https://files.pythonhosted.org/packages/d4/0c/9840c08189e030873387a73b90ada981885010dd9aea134d6de30cd24cb8/virtualenv-15.1.0.tar.gz"
+    sha256 "02f8102c2436bb03b3ee6dede1919d1dac8a427541652e5ec95171ec8adbc93a"
+  end
+
+  resource "virtualenv-clone" do
+    url "https://files.pythonhosted.org/packages/36/66/d95f83255138ded1aec242de5ee51536226fa7b2582a7ce0863fea5dc4f2/virtualenv-clone-0.2.6.tar.gz"
+    sha256 "6b3be5cab59e455f08c9eda573d23006b7d6fb41fae974ddaa2b275c93cc4405"
+  end
+
+  def install
+    # `pipenv` needs to be able to find `virtualenv` and `pew` on $PATH. Rather
+    # than installing symlinks for those scripts into /usr/local/bin we leave
+    # them in `#{libexec}/bin` and make /usr/local/bin/pipenv a wrapper script
+    # which adds `#{libexec}/bin` to $PATH.
+    venv = virtualenv_create(libexec)
+    venv.pip_install resources
+    venv.pip_install buildpath
+
+    # We add #{HOMEBREW_PREFIX}/bin to $PATH in front of #{libexec}/bin so that
+    # `pipenv --two` prefers the Homebrew installed version of python2 (if it
+    # exists) over the system one.
+    env = {
+      :PATH => "#{HOMEBREW_PREFIX}/bin:#{libexec}/bin:$PATH",
```

Comment:
The system itself doesn't have a python2 so I don't think I see the point in favoring Homebrew's python2 over some other python2 in the user's PATH.

Reply:
There's a python2 in `#{libexec}/bin` which is a virtualenv copy of the system python. I'll make the comment clearer.

I like being able to select Homebrew python with `--two` rather than `--python /usr/local/bin/python2` but it's a minor thing. Do you still want me to drop `#{HOMEBREW_PREFIX}/bin:` from the PATH?

I'm now wondering if

```
:PATH => "$PATH:#{libexec}/bin",
```

would be a better idea. That would respect the user's PATH for finding their preferred versions of python, virtualenv and pew with the versions in #{libexec}/bin as a fall back. However, if the user manually installs incompatible versions of  virtualenv or pew it would break pipenv. I'm not sure how stable their command line interface is.

Reply:
I found a different way to approach this. Rather than add anything else to $PATH I just remove `python2` and `python2.7` from `#{libexec}/bin`